programs/ziti-edge-tunnel/log_utils.c

```diff
@@ -0,0 +1,308 @@
+/*
+ Copyright 2019-2021 NetFoundry Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <ziti/ziti_log.h>
+#include <time.h>
+#include <file-rotator.h>
+#if _WIN32
+#include "windows/windows-service.h"
+#include <direct.h>
+#endif
+
+#if _WIN32
+#define MAXPATHLEN MAX_PATH
+#else
+#define MAXPATHLEN PATH_MAX
+#endif
+
+static FILE *ziti_tunneler_log = NULL;
+static uv_check_t *log_flusher;
+static struct tm *start_time;
+char* log_filename;
+static bool multi_writer = false;
+static const char* log_filename_base = "ziti-tunneler.log";
+static int rotation_count = 7;
+
+static char* get_log_filename() {
+    char curr_path[FILENAME_MAX]; //create string buffer to hold path
+#if _WIN32
+    _getcwd( curr_path, FILENAME_MAX );
+#else
+    sprintf(curr_path, "%s", "/tmp");
+#endif
+
+    char log_path[FILENAME_MAX];
+    sprintf(log_path, "%s/logs", curr_path);
+    int check;
+#if _WIN32
+    check = mkdir(log_path);
+#else
+    check = mkdir(log_path, 0755);
+#endif
+    if (check == 0) {
+        printf("\nlog path is created at %s", curr_path);
+    } else {
+        printf("\nlog path is found at %s", curr_path);
+    }
+
+    char time_val[32];
+    strftime(time_val, sizeof(time_val), "%Y%m%d0000", start_time);
+
+    char* log_filename = calloc(FILENAME_MAX, sizeof(char));
```

Comment:
this variable shadows the one at file scope, so the `sprintf` that follows isn't seen by functions that are using the global variable.